packages/frontend/src/components/filters/Filters.js

```diff
@@ -0,0 +1,272 @@
+import { useCallback, useEffect, useRef } from 'react'
+import { Button, useDisclosure } from '@chakra-ui/react'
+import { useFilters } from '../../hooks'
+import { MultiSelectFilter, InputFilter, RangeFilter, FilterGroup, ActiveFilters, IdentityFilter } from './'
+import { BottomSheet } from '../ui/sheets'
+import { ChevronIcon } from '../ui/icons'
+import { MultiLevelMenu } from '../ui/menus'
+import { MobileFilterMenu } from './MobileFilterMenu'
+import './Filters.scss'
+
+export const Filters = ({
+  filtersConfig,
+  initialFilters = {},
+  onFilterChange,
+  isMobile = false,
+  buttonText = 'Add filter',
+  className = ''
+}) => {
+  const defaultFilters = Object.fromEntries(
+    Object.keys(filtersConfig).map(key => [
+      key,
+      initialFilters?.[key] ?? filtersConfig[key].defaultValue
+    ])
+  )
+
+  const {
+    filters,
+    setFilters,
+    handleFilterChange: baseHandleFilterChange,
+    handleMultipleValuesChange: baseHandleMultipleValuesChange
+  } = useFilters(defaultFilters)
+
+  const previousFilters = useRef(filters)
+
+  const applyFilters = useCallback(() => {
+    if (typeof onFilterChange !== 'function') return
+
+    const processedFilters = (() => {
+      return Object.entries(filters).reduce((result, [key, value]) => {
+        const filterKeyConfig = filtersConfig[key]
+
+        if (filterKeyConfig.type === 'multiselect' && (filterKeyConfig.isAllSelected(value) || value?.length === 0)) {
+          return result
+        }
+
+        if (value === '') {
+          return result
+        }
+
+        if (filterKeyConfig.type === 'range') {
+          if (!value.min && !value.max) return result
+
+          if (value.min) result[`${key}_min`] = value.min
+          if (value.max) result[`${key}_max`] = value.max
+          return result
+        }
+
+        result[key] = value
```

Comment:
Map / Reduce functions prefer returning new objects instead of mutating the arguments